File: src/modeler/components/dispose.py
from sys import maxsize

from src.modeler.components.element import Element


class Dispose(Element):
    def __init__(self) -> None:
        super().__init__(0, 0)
        self.put_tnext(maxsize)

    def in_act(self) -> None:
        self.out_act()

    def out_act(self) -> None:
        super().out_act()

    def get_summary(self) -> str:
        nearest_tnext = self.get_tnext()
        if nearest_tnext != maxsize:
            self.average_load = self.quantity / nearest_tnext

        return f"\n##### {self.name} #####\n" f"quantity: {self.quantity}"

File: project-rust/src/modeler/components/dispose.rs
use rand::rngs::SmallRng;

use crate::modeler::components::element::Element;

pub fn in_act(e: &mut Element, rng: &mut SmallRng) {
    e.out_act(rng);
}

pub fn out_act(e: &mut Element) {
    e.quantity += 1;
}

pub fn get_summary(e: &Element) -> String {
    format!(
        "\n
        ##### {} #####\n
        quantity = {}",
        e.name, e.quantity
    )
}
